### DISCUSSION TITLE
Google is choosing a Deployment Preview URL as canonical instead of my current domain URL.

### DISCUSSION THREAD

**Discussion by jon-minimal** [vercel/community#5843] [Help]
### Summary

Hey,

I am trying submit my page to Google in order to be indexed, but they are choosing a Vercel Deployment Preview URL as canonical instead of my current domain URL.

Why is that? Is there any way to prevent it?

Please find attached a screenshot from Google Search Console result.

Thanks in advance, let me know if you need any other info from my part.

### Steps to Reproduce

<img width="746" alt="Screenshot 2024-02-15 at 1 58 56 PM" src="https://github.com/vercel/community/assets/160207101/f7743d76-962c-48bb-a59d-7c8efd232f61">

**Comment by Jempan** [comment]
I have the exact same problem, did you find a solution?

**Reply by versecafe** [reply]
Make sure robots.txt when not on your production domain is set to no index and have the sitemap on all your domains always point to the domain you want, those two combined tend to fix the problem

**Reply by Jempan** [reply]
Thanks for your suggestions! Vercel automatically attach the no index header to deployment domains but it seems that googlebot does not respect it. Also I have a sitemap and canonical links set on every page but still the Google-selected canonical is pointing to a vercel deployment url for a couple important pages including my main landing page :(
